File: src/test/java/io/tripled/social/client/presentation/TestOutput.java
package io.tripled.social.client.presentation;

import org.hamcrest.Matcher;

import java.util.LinkedList;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.empty;
import static org.hamcrest.Matchers.hasItems;

public class TestOutput implements Output {

  private final LinkedList<String> outputs;

  public TestOutput() {
    this.outputs = new LinkedList<>();
  }

  @Override
  public void print(String message) {
    this.outputs.add(message);
  }

  public void assertContains(Matcher<String>... message) {
    assertThat(this.outputs, hasItems(message));
  }

  void assertIsEmpty() {
    assertThat(this.outputs, empty());
  }

}

File: src/test/kotlin/io/tripled/social/client/presentation/TestOutput.kt
package io.tripled.social.client.presentation

import org.hamcrest.Matcher
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import java.util.*

class TestOutput : Output {
    private val outputs: LinkedList<String> = LinkedList<String>()

    override fun print(message: String) {
        this.outputs.add(message)
    }

    fun assertContains(vararg message: Matcher<String>) {
        MatcherAssert.assertThat(this.outputs, Matchers.hasItems(*message))
    }

    fun assertIsEmpty() {
        MatcherAssert.assertThat(this.outputs, Matchers.empty())
    }
}
